src/udiskslinuxpartitiontable.c

```diff
@@ -118,15 +118,77 @@ udisks_linux_partition_table_update (UDisksLinuxPartitionTable  *table,
                                      UDisksLinuxBlockObject     *object)
 {
   const gchar *type = NULL;
-  UDisksLinuxDevice *device = NULL;;
+  UDisksLinuxDevice *device = NULL;
+  UDisksDaemon *daemon = NULL;
+  guint num_parts = 0;
+  const gchar **partition_object_paths = NULL;
+  GList *partition_objects = NULL;
+  GList *object_p = NULL;
+  guint i = 0;
 
+  /* update partition table type */
   device = udisks_linux_block_object_get_device (object);
   if (device != NULL)
     type = g_udev_device_get_property (device->udev_device, "ID_PART_TABLE_TYPE");
 
   udisks_partition_table_set_type_ (UDISKS_PARTITION_TABLE (table), type);
 
+  /* update list of partitions */
+  daemon = udisks_linux_block_object_get_daemon (UDISKS_LINUX_BLOCK_OBJECT (object));
+
+  partition_objects = udisks_linux_partition_table_get_partitions (daemon, UDISKS_PARTITION_TABLE (table));
+  num_parts = g_list_length (partition_objects);
```

Comment:
Would be nicer if ``udisks_linux_partition_table_get_partitions()`` had an extra out argument for returning the number of partitions.